src/lib/integration_test_lib/wait_condition.ml

```diff
@@ -118,8 +118,17 @@ struct
     ; hard_timeout = Slots (soft_timeout_in_slots * 2)
     }
 
-  let payment_to_be_included_in_frontier ~sender_pub_key ~receiver_pub_key
-      ~amount =
+  type command_type = Send_payment | Send_delegation
+
+  let command_type_to_string command_type =
+    match command_type with
+    | Send_payment ->
+        "Send Payment"
+    | Send_delegation ->
+        "Send Delegation"
+
+  let signed_command_to_be_included_in_frontier ~sender_pub_key
```

Comment:
In the future, we should simplify this implementation by logging the transaction hash and checking that instead of attempting to check all the predicates to see if the transaction was equivalent.

Reply:
It concerns me that the current implementation isn't comparing the nonce of the commands. Can we at least add that for now, in lieu of a txn hash implementation?